Add more configuration options for helm chart

Aakcht opened the pull request:
This PR adds possibility to specify affinity/tolerations/securityContext/nodeSelector/labels in helm chart for main Volcano pods(scheduler/admission/controller).

resolves #2779

Aakcht commented:
Looks like check fail is not related to the changes of this PR. @wangyang0616 , please, take a look.

wangyang0616 commented:
> Looks like check fail is not related to the changes of this PR. @wangyang0616 , please, take a look.

See issue: [#2777](https://github.com/volcano-sh/volcano/issues/2777), this issue has proposed a repair PR ([#2780](https://github.com/volcano-sh/volcano/pull/2780)), and it should be able to solve the current CI problem after being merged

wangyang0616 commented:
priority/important-soon

wangyang0616 commented:
/priority important-soon

Aakcht commented:
The repair PR (https://github.com/volcano-sh/volcano/pull/2780) helped, after rebasing the checks are successful.

william-wang reviewed (approved):
/lgtm

william-wang commented:
@Aakcht please help to handle the confliction:)

Aakcht commented:
Hi, @william-wang . As I see, there was this PR #2836 that was merged and it added nodeSelector & affinity & tolerations(it caused the conflicts). I rebased and reverted the changes from that PR since I think it would be better to be able to set  nodeSelector/affinity/tolerations per component also(as this PR). If you think it would be better to change default nodeSelector/affinity/tolerations parameter names in accordance with #2836, please tell me.

william-wang commented:
@Aakcht Agree with you, it's better to configure affinity/toleration/selector per component. Please make the ci happy and we can merge this pr.

Aakcht commented:
@william-wang looks like I'll need help with the CI. 
* It seems `Code Verify / Verify codes, generated files (pull_request)` check failure is not related to this PR - I can see the same error in master branch.
* `E2E Spark Integration Test / E2E about Spark Integration test (pull_request)` I don't see any errors in the logs with this one, looks like it simply failed because of timeout. I suspect rerunning the check might help, but as I see this check is not required, so I think the previous check error should be fixed first.

wangyang0616 commented:
> @william-wang looks like I'll need help with the CI.
> 
> * It seems `Code Verify / Verify codes, generated files (pull_request)` check failure is not related to this PR - I can see the same error in master branch.
> * `E2E Spark Integration Test / E2E about Spark Integration test (pull_request)` I don't see any errors in the logs with this one, looks like it simply failed because of timeout. I suspect rerunning the check might help, but as I see this check is not required, so I think the previous check error should be fixed first.

@Aakcht Master's code verify encountered an error and has been repaired, refer to pr: [fix goimport check error #2849](https://github.com/volcano-sh/volcano/pull/2849)

Aakcht commented:
Hello, @wangyang0616 , @william-wang I rebased the PR, now all the checks are successful.

Aakcht commented:
Hi, @william-wang, I rebased the PR one more time case there were new merge conflicts - now all should be good, please take a look.

jacobsalway commented:
Hey all, just adding that this PR would ease the Helm chart configuration for my specific deployment circumstance. I need to add tolerations and node selectors to all deployment components, but https://github.com/volcano-sh/volcano/pull/2836 on master is missing the configurability on the `admission-init` job, which this PR adds.

Aakcht commented:
Hi, @hwdef @wangyang0616 @Thor-wl @william-wang , looking for review/aproval/merge.

hwdef commented on installer/volcano-development.yaml:
```diff
@@ -153,6 +153,8 @@ spec:
           image: volcanosh/vc-webhook-manager:latest
           imagePullPolicy: Always
           name: admission
+          resources:
+            null
```
what caused this

Aakcht replied:
@hwdef , I suspect it happened because these to files([1](https://github.com/volcano-sh/volcano/blob/master/hack/generate-yaml.sh), [2](https://github.com/volcano-sh/volcano/blob/master/hack/check-generated-yaml.sh)) that compare deployments generated by `helm template` with [volcano deployment](https://github.com/volcano-sh/volcano/blob/master/installer/volcano-development.yaml). 
So since I added empty resources by default the rendered files by `helm template` have null for resources.


If needed I can change the PR so resources will not be present at all when not specified, so please tell me if you think I should change it. But I'm not sure it's needed, since in the end the result is the same in Kubernetes.

hwdef commented:
Please close PR and reopen it to retrigger CI

Aakcht commented:
@hwdef I reopened it, but the failed check remained and I'm pretty sure that the failed check is not related to this PR (You can also see the same error in the latest opened PR #2920) . 

Also it looks like `E2E about Spark Integration test` check is not mandatory since it does not have `required` tag.

hwdef commented:
@wangyang0616 Please check this CI failed.

Aakcht commented:
Hi, @hwdef @wangyang0616 @Thor-wl @william-wang , please take a look, the checks are now successful.

hwdef reviewed (commented):
/lgtm
